Games/types/Mafia/roles/cards/GainKnifeIfVisitedNonMonster.js

```diff
@@ -14,6 +14,9 @@ module.exports = class GainKnifeIfVisitedNonMonster extends Card {
                     if (this.game.getStateName() !== "Night")
                         return;
 
+                    if (!this.player.alive)
```

Comment:
accursed doll has two abilities - learn roles and get knife.

just to check it's intended to continue learning roles when dead

Reply:
it learns roles while alive, fixed